### DISCUSSION TITLE
Slow initial requests to api endpoint with refreshInterval is used

### DISCUSSION THREAD

**Discussion by joepagan** [vercel/swr#860] [Q&A] [answered]
Hey,

I would like my app to make a request to an endpoint on the initial page load, then every 10 seconds.

I presumed setting `refreshInterval: 10000,` would be perfect for this however I have an issue where:

- first request can take >10s to be made before `data` is available
- when removing `refreshInterval` prop, it also takes a long time >10s
- when settings `refreshInterval` to something small `1000` the data loads very fast.

Can someone explain if I am doing something wrong here?

Example:

```js
const { data, err } = useSWR('/api/data', fetcher, {
    refreshInterval: 1000,
    initialData: {
      distance: 0,
      members: 0,
      activities: 0,
    },
  });
```

Later I pass to a component like so:

```jsx
   <Stat
       value={data ? data.distance : 0}
    />
```

I have also tried using a preload, in this scenario I can see the request in the network tab in browser instantly, however, `data` does not appear to have updated with this information until >10s also:

```html
<link rel="preload" href="/api/data" as="fetch" crossorigin="anonymous" />
```

I would very much appreciate any suggestions on how to proceed
TIA

**Comment by huozhi** [comment]
`refreshInterva` is the interval between 2 revalidations, if your fetcher might take 10s, the next revalidation will happen after `10 * 1000 + refreshInterval`.

```
--- 1st revalidation --- refresh interval duration --- 2nd revalidation
```

**Reply by joepagan** [reply]
Forgive me, I presume by revalidation you mean a successful request?

I want `data` to be available as soon as possible, then to check & update `data` on a 10second interval.

In my initial post I tried to highlight that a high `refreshInterval` means that my first request is slow as hell (but shouldn't be)

**Reply by joepagan** [reply]
Example from my firefox network tab, you can see the first request is 10s in under the "Response Time" column:

![image](https://user-images.githubusercontent.com/2337910/103636991-b5b66f00-4f42-11eb-9440-d9c925ee337e.png)

I wish for the requests to be:
0s, 10s, 20s etc...

**Comment by huozhi** [comment, marked as answer]
> I presume by revalidation you mean a successful request?
Yes! and also inlcude the failed ones I think. it's the procedure that executes the fetcher and waits for the result.

you can use `revalidateOnMount` config for first revalidation. and still keep the `refreshInterval` for regular checkings.

```diff
const { data, err } = useSWR('/api/data', fetcher, {
+   revalidateOnMount: true,
    refreshInterval: 1000,
    initialData: {
      distance: 0,
      members: 0,
      activities: 0,
    },
  });
```

**Reply by joepagan** [reply]
Exactly what I wanted, thanks so much @huozhi !
